cmd/backup-manager/app/backup/backup.go

```diff
@@ -0,0 +1,224 @@
+}
+
+func (bo *BackupOpts) getTikvGCLifeTime(db *sql.DB) (string, error) {
+	var tikvGCTime string
+	row := db.QueryRow("select variable_value from ? where variable_name= ?", constants.TidbMetaTable, constants.TikvGCVariable)
+	err := row.Scan(&tikvGCTime)
+	if err != nil {
+		return tikvGCTime, fmt.Errorf("query cluster %s %s failed, err: %v", bo, constants.TikvGCVariable, err)
+	}
+	return tikvGCTime, nil
+}
+
+func (bo *BackupOpts) setTikvGClifeTime(db *sql.DB, gcTime string) error {
+	_, err := db.Exec("update ? set variable_value = ? where variable_name = ?", constants.TidbMetaTable, gcTime, constants.TikvGCVariable)
+	if err != nil {
+		return fmt.Errorf("set cluster %s %s failed, err: %v", bo, constants.TikvGCVariable, err)
+	}
+	return nil
+}
+
+func (bo *BackupOpts) dumpTidbClusterData() (string, error) {
+	bfPath := bo.getBackupFullPath()
+	err := util.EnsureDirectoryExist(bfPath)
+	if err != nil {
+		return "", err
+	}
+	args := []string{
+		fmt.Sprintf("--outputdir=%s", bfPath),
+		fmt.Sprintf("--host=%s", bo.TidbSvc),
+		"--port=4000",
+		fmt.Sprintf("--User=%s", bo.User),
+		fmt.Sprintf("--Password=%s", bo.Password),
+		"--long-query-guard=3600",
+		"--tidb-force-priority=LOW_PRIORITY",
+		"--verbose=3",
+	}
+
+	dumper := exec.Command("/mydumper", args...)
+	if err := dumper.Start(); err != nil {
+		return bfPath, fmt.Errorf("cluster %s, start mydumper command %v falied, err: %v", bo, args, err)
+	}
+	if err := dumper.Wait(); err != nil {
+		return bfPath, fmt.Errorf("cluster %s, execute mydumper command %v failed, err: %v", bo, args, err)
+	}
+	return bfPath, nil
+}
+
+func (bo *BackupOpts) backupDataToRemote(source, bucketURI string) error {
+	destBucket := util.NormalizeBucketURI(bucketURI)
+	tmpDestBucket := fmt.Sprintf("%s.tmp", destBucket)
+	// TODO: We may need to use exec.CommandContext to control timeouts.
+	rcCopy := exec.Command("rclone", constants.RcloneConfigArg, "copyto", source, tmpDestBucket)
+	if err := rcCopy.Start(); err != nil {
+		return fmt.Errorf("cluster %s, start rclone copyto command for upload backup data %s falied, err: %v", bo, bucketURI, err)
+	}
+	if err := rcCopy.Wait(); err != nil {
+		return fmt.Errorf("cluster %s, execute rclone copyto command for upload backup data %s failed, err: %v", bo, bucketURI, err)
+	}
+
+	glog.Info("upload cluster %s backup data %s successfully, now move it to permanent URL", bo, bucketURI)
+
+	// the backup was a success
+	// remove .tmp extension
+	rcMove := exec.Command("rclone", constants.RcloneConfigArg, "moveto", tmpDestBucket, destBucket)
+
+	if err := rcMove.Start(); err != nil {
+		return fmt.Errorf("cluster %s, start rclone moveto command falied, err: %v", bo, err)
+	}
+
+	if err := rcMove.Wait(); err != nil {
+		return fmt.Errorf("cluster %s, execute rclone moveto command falied, err: %v", bo, err)
+	}
+	return nil
+}
+
+func (bo *BackupOpts) cleanRemoteBackupData(bucket string) error {
```

Comment:
do we need to clean local backup data?

Reply:
I will add the cleanup logic of local backup data after the manual test passes